packages/flutter/lib/src/cupertino/text_field.dart

```diff
@@ -541,12 +553,35 @@ class _CupertinoTextFieldState extends State<CupertinoTextField> with AutomaticK
   }
 
   void _handleSingleLongTapEnd(LongPressEndDetails details) {
-    _editableTextKey.currentState.showToolbar();
+    if (_isSelectionOverlayEnabled)
+      _editableText.showToolbar();
   }
 
   void _handleDoubleTapDown(TapDownDetails details) {
     _renderEditable.selectWord(cause: SelectionChangedCause.tap);
-    _editableTextKey.currentState.showToolbar();
+    if (_isSelectionOverlayEnabled)
+      _editableText.showToolbar();
+  }
+
+  bool _shouldShowHandles(SelectionChangedCause cause) {
+    // When the text field was triggered by anything other than a touch screen,
+    // we shouldn't show the selection handles.
+    // If [_deviceKind] is null we still show the handles for backwards-compatibility
+    // reasons.
+    if (!_isSelectionOverlayEnabled)
+      return false;
+    // On iOS, we don't show handles when the selection is collapsed.
+    if (_effectiveController.selection.isCollapsed)
+      return false;
+    if (cause == SelectionChangedCause.keyboard)
+      return false;
+
```

Comment:
The reason I formatted them this way is because the first two return false while the 3rd and 4th return true. I'll add an empty line between all of them.